tests/tests_twobody/test_perturbations.py

```diff
@@ -221,6 +221,51 @@ def test_atmospheric_drag():
     )
 
 
+@pytest.mark.slow
+def test_atmospheric_demise():
+    # Test an orbital decay that hits Earth. No analytic solution.
+    R = Earth.R.to(u.km).value
+
+    orbit = Orbit.circular(Earth, 230 * u.km)
+
+    # parameters of a body
+    C_D = 2.2  # dimentionless (any value would do)
+    A = ((np.pi / 4.0) * (u.m ** 2)).to(u.km ** 2).value  # km^2
+    m = 100  # kg
+
+    # parameters of the atmosphere
+    rho0 = rho0_earth.to(u.kg / u.km ** 3).value  # kg/km^3
+    H0 = H0_earth.to(u.km).value  # km
+
+    tofs = [365] * u.d  # actually hits the ground a bit after day 48
+
+    def event_lithobrake(R, t, u):
+        # search for where H - R crosses zero, and then terminate
+        H = norm(u[:3])
+        return H - R
+
+    event_lithobrake_r = functools.partial(event_lithobrake, R)
+    event_lithobrake_r.terminal = True
```

Comment:
> You can assign attributes like `event.terminal = True` to any function in Python.

[Ughhhhh](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html).

I see two options here:

* Make the event a class:

```python
class LithobrakeEvent:
    def __init__(self, R):
        self._R = R

    @property
    def terminal(self):
        return True

    def __call__(self, t, u):
        H = norm(u[:3])
        return H - self._R

events = [LithobrakeEvent(R)]
```

* Add a `# type: ignore` in the same line